File: docs/guides/ruby/searchFuturePublishedBooks.rb
import(time)

require "algolia"

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

# Get current timestamp
date_timestamp = Time.now.to_i
search_params = Algolia::Search::SearchParamsObject.new(
  query: "<YOUR_SEARCH_QUERY>",
  filters: "date_timestamp > #{date_timestamp}"
)

client.search_single_index("<YOUR_INDEX_NAME>", search_params)

File: docs/guides/python/searchFuturePublishedBooks.py
import time

from algoliasearch.search.client import SearchClientSync


_client = SearchClientSync("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")


date_timestamp = time.time()  # Get current timestamp
search_params = {
    "query": "<YOUR_SEARCH_QUERY>",
    "filters": f"date_timestamp > {date_timestamp}",
}

_client.search_single_index(
    index_name="<YOUR_INDEX_NAME>",
    search_params=search_params,
)
